ledger/src/bank_forks_utils.rs

```diff
@@ -61,16 +67,57 @@ pub fn load(
         accounts_update_notifier,
     );
 
-    blockstore_processor::process_blockstore_from_root(
+    /// maximum drop bank signal queue length
+    /// (consider de-duping this const and `solana-core::validator::MAX_DROP_BANK_SIGNAL_QUEUE_SIZE`)
+    const MAX_DROP_BANK_SIGNAL_QUEUE_SIZE: usize = 10_000;
+
+    // There should only be one bank, the root bank in BankForks. Thus all banks added to
+    // BankForks from now on will be descended from the root bank and thus will inherit
+    // the bank drop callback.
+    assert_eq!(bank_forks.read().unwrap().banks().len(), 1);
+    let (pruned_banks_sender, pruned_banks_receiver) =
+        crossbeam_channel::bounded(MAX_DROP_BANK_SIGNAL_QUEUE_SIZE);
+    {
+        let root_bank = bank_forks.read().unwrap().root_bank();
+        root_bank.set_callback(Some(Box::new(
+            root_bank
+                .rc
+                .accounts
+                .accounts_db
+                .create_drop_bank_callback(pruned_banks_sender),
+        )));
+    }
+
+    let abs_request_handler = AbsRequestHandler {
+        snapshot_request_handler: None,
+        pruned_banks_receiver,
+    };
+    let exit = Arc::new(AtomicBool::new(false));
+
+    let accounts_background_service = AccountsBackgroundService::new(
+        bank_forks.clone(),
+        &exit,
+        abs_request_handler,
+        false,
```

Comment:
This line is what prevented this from working; this `false` corresponds to `accounts_db_caching_enabled` arg, so cleanup wasn't happening as expected.